src/sql/workbench/services/connection/browser/connectionDialogWidget.ts

```diff
@@ -392,10 +397,24 @@ export class ConnectionDialogWidget extends Modal {
 	}
 
 	private onConnectionClick(element: IConnectionProfile, connect: boolean = false): void {
-		if (connect) {
-			this.connect(element);
-		} else {
-			this._onFillinConnectionInputs.fire(element);
+		const isProviderAvailable = this._capabilitiesService.providers[element.providerName] !== undefined;
+		if (isProviderAvailable) {
+			if (connect) {
+				this.connect(element);
+			} else {
+				this._onFillinConnectionInputs.fire(element);
+			}
+		}
+		else {
+			this._notificationService.prompt(Severity.Error,
+				localize('connectionDialog.connectionProviderNotInstalled', "The extension that supports provider type: '{0}' is not currently installed. Please install it and try again.", element.providerName),
```

Comment:
It's not ideal, but since PG is the main external provider we support what about doing a hard check for them here and providing a direct link to install? Would make their experience better and if we start having more later then we can look into refactoring to support doing the lookup

Reply:
(and if it's not PG then keep what you have currently so it still does something useful)

Reply:
this is requested by another partner team.
I guess it is a fair compromise to have the provider-extension name mapping here. what do you think?

Reply:
Fine by me, it's probably more important to have a decent experience here and will be easy enough to take out later if we need to.

Reply:
updated the PR description and added the gif for the new scenario